Tests/FluentAssertions.Specs/Specialized/AssemblyAssertionSpecs.cs

```diff
@@ -260,6 +260,79 @@ public void When_an_assembly_is_null_and_Should_BeNull_is_asserted_it_should_suc
             act.Should().NotThrow();
         }
     }
+
+    public class BeUnsigned
+    {
+        [Fact]
+        public void Guards_for_unsigned_assembly()
+        {
+            // Arrange
+            var unsignedAssembly = typeof(UnsignedAssembly.Class).Assembly;
+
+            // Act & Assert
+            unsignedAssembly.Should().BeUnsigned();
+        }
+
+        [Fact]
+        public void Throws_for_signed_assembly()
+        {
+            // Arrange
+            var signedAssembly = typeof(AssertionOptions).Assembly;
+
+            // Act
+            Action act = () => signedAssembly.Should().BeUnsigned();
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("Did not expect the assembly * to be signed, but it is.");
+        }
+    }
+
+    public class HavePublicKey
+    {
+        [Fact]
+        public void Guards_for_signed_assembly_with_expected_public_key()
+        {
+            // Arrange
+            var signedAssembly = typeof(AssertionOptions).Assembly;
+
+            // Act & Assert
+            signedAssembly.Should().HavePublicKey(
+                "0024000004800000940000000602000000240000525341310004000001000100" +
+                "2D25FF515C85B13BA08F61D466CFF5D80A7F28BA197BBF8796085213E7A3406F" +
+                "970D2A4874932FED35DB546E89AF2DA88C194BF1B7F7AC70DE7988C78406F762" +
+                "9C547283061282A825616EB7EB48A9514A7570942936020A9BB37DCA9FF60B77" +
+                "8309900851575614491C6D25018FADB75828F4C7A17BF2D7DC86E7B6EAFC5D8F");
+        }
+
+        [Fact]
+        public void Throws_for_unsigned_assembly()
+        {
+            // Arrange
+            var unsignedAssembly = typeof(UnsignedAssembly.Class).Assembly;
```

Comment:
If @dennisdoomen is also fine with it, here's a dummy implementation of an `Assembly` to avoid adding `UnsignedAssembly`.

```cs
public class MyAssembly : Assembly
{
    private readonly AssemblyName assemblyName = new();

    public override string FullName => "MyAssembly";

    public MyAssembly(string publicKey)
    {
        assemblyName.SetPublicKey(FromHexString(publicKey));
    }

    public override AssemblyName GetName() => assemblyName;

#if NET6_0_OR_GREATER
    private static byte[] FromHexString(string chars) => Convert.FromHexString(chars);
#else
    private static byte[] FromHexString(string chars)
    {
        var bytes = new byte[chars.Length / 2];

        for (int i = 0, j = 0; j < bytes.Length; j++, i += 2)
        {
            bytes[j] = (byte)((Hex(chars[i]) << 4) | Hex(chars[i + 1]));

            static byte Hex(char c) => (byte)(c > '9' ? (c & 0x5F) - 55 : c - '0');
        }

        return bytes;
    }
#endif
}
```